### PR TITLE
Add JPMS descriptor

### PR DESCRIPTION
**Pull request by Thihup**
Signed-off-by: Thiago Henrique Hüpner <thihup@gmail.com>

### CODE REVIEW

**Comment by gregw** [comment]
@sbordet can you please review this

File: api/pom.xml [hunk 1]
```diff
@@ -257,6 +269,11 @@
                         </goals>
                         <configuration>
                             <source>1.8</source>
+                            <release>8</release>
+                            <sourceFileExcludes>
+                                <sourceFileExclude>module-info.java</sourceFileExclude>
+                            </sourceFileExcludes>
+                            <notimestamp>true</notimestamp>
```
Comment (joakime) [inline api/pom.xml:276]: This configuration on the javadoc-plugin seems out of scope for this PR.

File: api/pom.xml [hunk 2]
```diff
@@ -135,12 +135,27 @@
...
+                    </compilerArgs>
                 </configuration>
+                <executions>
+                    <execution>
+                        <id>base-compile</id>
+                        <goals>
+                            <goal>compile</goal>
+                        </goals>
+                        <configuration>
+                            <release>8</release>
```
Comment (gregw) [inline api/pom.xml:152]: I don't think we need or should build with java8.   That will only result in different builds for the same version of the api.

File: api/src/main/java/module-info.java [hunk 3]
```diff
@@ -0,0 +1,6 @@
+module jakarta.servlet {
+    exports jakarta.servlet;
+    exports jakarta.servlet.annotation;
+    exports jakarta.servlet.descriptor;
+    exports jakarta.servlet.http;
```
Comment (gregw) [inline api/src/main/java/module-info.java:5]: Jetty has been adding JPMS to our re-bundling of 4.0 API and we have included an export for jakarta.servlet.resources which includes all the XSDs and DTDs.    I believe tomcat also bundles the resources the same way.

I don't think we really want to have another JPMS module involved that has the XSDs and DTDs... if so should this module depend on that?

File: api/pom.xml [hunk 4]
```diff
@@ -214,9 +229,6 @@
                 <configuration>
                     <archive>
                         <manifestFile>${project.build.outputDirectory}/META-INF/MANIFEST.MF</manifestFile>
```
Comment (gregw) [inline api/pom.xml:231]: Probably out of scope for this PR, but we really should also add the OSGi headers in the manifest.

File: api/pom.xml (hunk 2 above)
Comment (joakime) [reply api/pom.xml:152]: I thought all Jakarta EE9 spec artifacts were supposed to be Java 8 bytecode.

File: api/pom.xml (hunk 4 above)
Comment (joakime) [reply api/pom.xml:231]: The `/api/` build here already adds OSGi manifest headers.

The result of `mvn clean install` on `master` branch today.

```
Manifest-Version: 1.0
Archiver-Version: Plexus Archiver
Created-By: Apache Maven Bundle Plugin
Built-By: joakim
Build-Jdk: 11.0.6
Automatic-Module-Name: java.servlet
Bnd-LastModified: 1586271343407
Bundle-Description: Jakarta Servlet 5.0
Bundle-DocURL: https://www.eclipse.org
Bundle-License: http://www.eclipse.org/legal/epl-2.0, https://www.gnu.or
 g/software/classpath/license.html
Bundle-ManifestVersion: 2
Bundle-Name: Jakarta Servlet
Bundle-SymbolicName: jakarta.servlet-api
Bundle-Vendor: Eclipse Foundation
Bundle-Version: 5.0.0
Export-Package: jakarta.servlet;uses:="jakarta.servlet.annotation,jakart
 a.servlet.descriptor";version="5.0.0",jakarta.servlet.annotation;uses:=
 "jakarta.servlet";version="5.0.0",jakarta.servlet.descriptor;version="5
 .0.0",jakarta.servlet.http;uses:="jakarta.servlet";version="5.0.0"
Extension-Name: jakarta.servlet
Implementation-Vendor: Eclipse Foundation
Implementation-Vendor-Id: org.eclipse
Implementation-Version: 5.0.0-SNAPSHOT
Import-Package: jakarta.servlet,jakarta.servlet.annotation,jakarta.servl
 et.descriptor
Require-Capability: osgi.ee;filter:="(&(osgi.ee=JavaSE)(version=1.8))"
Specification-Vendor: Eclipse Foundation
Specification-Version: 5.0
Tool: Bnd-4.0.0.201805111645
```

File: api/pom.xml (hunk 2 above)
Comment (joakime) [reply api/pom.xml:152]: I just checked the master branch of jsp-api, jaxb-api, and cdi specs.
They all target `<release>1.8</release>` on their compile.

Comment (gregw) [reply api/pom.xml:152]: So is EE9 is java8 byte code.... and this PR excludes module-info.java if built with java8, then this is a noop?  Or am I miss reading this?

Comment (joakime) [reply api/pom.xml:152]: There were 2 `maven-compiler-plugin` entries in this PR when it was first submitted.

1. First one did a normal bytecode compile on release 8 without `module-info.java`.
2. Next one did a compile that only included `module-info.java` using release 9.

The later `package` step bundled them up into a single jar.

File: api/src/main/java/module-info.java (hunk 3 above)
Comment (Thihup) [reply api/src/main/java/module-info.java:5]: Probably. I'm not sure how to proceed right now. 
Could you please give me a hint in what I should look next?

File: api/pom.xml (hunk 2 above)
Comment (Thihup) [reply api/pom.xml:152]: I haven't changed the build configuration since the first time I've submitted.
It's actually pretty close to what is done in [JPA api build](https://github.com/eclipse-ee4j/jpa-api/blob/master/api/pom.xml#L254)

File: api/src/main/java/module-info.java (hunk 3 above)
Comment (gregw) [reply api/src/main/java/module-info.java:5]: yeah, it's not something that you can do on your own here.  It is a policy decision that has to be made at least by the project, but more likely by the EE community as a whole.   So I think it's a bit of a road block, but unfortunately due to the nature of JPMS modules, we really don't want to get it wrong and then try to modify later.  The first published non auto module name needs to be pretty correct in content and intent.    

@stuartwdouglas @markt-asf @bshannon @dblevins any thoughts on this one?

Comment (bshannon) [reply api/src/main/java/module-info.java:5]: You shouldn't export resources that aren't defined by the spec.
The spec doesn't define a resource location for the schemas.
